src/main/java/org/junitpioneer/jupiter/json/ObjectMapperProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.json;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.json.JsonReadFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Service interface for providing a custom {@link com.fasterxml.jackson.databind.ObjectMapper} instance at runtime.
+ * The default implementation doesn't register any additional Jackson modules.
+ *
+ * @see com.fasterxml.jackson.databind.Module
+ */
+public interface ObjectMapperProvider {
+
+	ObjectMapper get();
+
+	default ObjectMapper getLenient() {
+		return get()
+				.copyWith(JsonFactory
```

Comment:
When creating the custom provider, in a newer version `2.15.2` of `jackson-databind`, the `copyWith` causes exception to be thrown:
```text
Failed copy()/copyWith(): com.fasterxml.jackson.databind.json.JsonMapper (version: 2.15.2) does not override copy()/copyWith(); it has to
java.lang.IllegalStateException: Failed copy()/copyWith(): com.fasterxml.jackson.databind.json.JsonMapper (version: 2.15.2) does not override copy()/copyWith(); it has to
	at com.fasterxml.jackson.databind.ObjectMapper._checkInvalidCopy(ObjectMapper.java:750)
	at com.fasterxml.jackson.databind.ObjectMapper.copyWith(ObjectMapper.java:738)
(...)
```
changing it to:
```
	default ObjectMapper getLenient() {
		return new JsonMapper(JsonFactory
				.builder()
				.enable(JsonReadFeature.ALLOW_UNQUOTED_FIELD_NAMES)
				.enable(JsonReadFeature.ALLOW_JAVA_COMMENTS)
				.enable(JsonReadFeature.ALLOW_SINGLE_QUOTES)
				.enable(JsonReadFeature.ALLOW_TRAILING_COMMA)
				.build());
	}
```
Solved the issue.
I created some [sample project](https://github.com/PrzemyslawSwiderski/spring-boot-playground) to recreate.

Reply:
You can overwrite `getLenient()`.

Originally, I was only thinking of `ObjectMapper` but I'll take another look.

Reply:
Right, I have to override it but users can be confused since it is not required and the exception message does not tell to do so

Reply:
Can't say I am very fond of it... but I do believe I came up with an acceptable solution.

Reply:
Let me know if you think it can be improved - I amended the `default` method to consider `JsonMapper` as a special use-case.

Reply:
Works great with my sample now. Thanks :)